mesonbuild/build.py

```diff
@@ -369,22 +372,22 @@ def __init__(self, name, subdir, subproject, build_by_default, for_machine: Mach
         if not hasattr(self, 'typename'):
             raise RuntimeError('Target type is not set for target class "{}". This is a bug'.format(type(self).__name__))
 
-    def __lt__(self, other: T.Any) -> T.Union[bool, type(NotImplemented)]:
+    def __lt__(self, other: object) -> T.Union[bool, type(NotImplemented)]:
```

Comment:
Does the `type(NotImplemented)` thing actually work? It looks like mypy is smart enough to understand that `-> bool`, so maybe just stick with that?

Reply:
I know I started the `NotImplemented` in return type thing...

Reply:
I just did a `s/other: T.Any/other: object/g` on all files and didn't check the return type. I am not sure if `type(NotImplemented)` works, but a straight `-> bool` would be IMHO better anyway.